website/content/en/docs/contribution-guidelines/plugins.md

```diff
@@ -0,0 +1,78 @@
+```
+
+The above example will scaffold a custom code in the controllers after when an api is created to allow the users develop solutions using the [kubebuilder declarative pattern][kubebuilder-declarative-pattern]. (e.g [default scaffold][default-scaffold] versus [example][kubebuilder-declarative-pattern-example]).
+
+Note that custom plugins which are called via the init sub-command, will be added in the `layout` field of the PROJECT file and will work such as global plugins. Then, it means that for any sub-command executed; they will also be called. 
+
+## Plugins Vision
+
+Maintainers and contributors are able to create their own plugin(s) using the same standards and approaches. That would allow these solutions to be theoretically compatible and useful for SDK CLI as well. Currently, SDK cannot discover and use plugins that are not ship/build with its binary. However, it has been discussed in the issue [kubebuilder/issues/1378][kb-issue].
+
+## How to create your own plugins
+
+If you are looking to develop similar solutions to allow users, for example, to create projects using other languages or that could work as helpers for the projects built with SDK/Kubebuilder projects, then see the [Creating your own plugins][create-your-own-plugins] to know how you can do that following this approach as its benefits.
+
+[kb-plugins-doc]: https://deploy-preview-2115--kubebuilder.netlify.app/plugins/plugins.html
+[kb-int-sdk]: https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/integrating-kubebuilder-and-osdk.md
+[kb-language-plugins]: https://deploy-preview-2115--kubebuilder.netlify.app/plugins/creating-plugins.html#language-based-plugins
+[kustomize]: https://github.com/kubernetes-sigs/kustomize
+[bundle]: https://github.com/operator-framework/operator-registry/blob/v1.16.1/docs/design/operator-bundle.md#operator-bundle
+[kb-project]: https://deploy-preview-2115--kubebuilder.netlify.app/plugins/project-config.html
+[sdk-cli-run-bundle]: ../../docs/cli/operator-sdk_run
+[project-layout]: ../../docs/overview/project-layout
+[plugin-manifest]: https://github.com/operator-framework/operator-sdk/tree/master/internal/plugins/manifests/v2
+[plugin-scorecard]: https://github.com/operator-framework/operator-sdk/tree/master/internal/plugins/scorecard/v2
+[kubebuilder-declarative-pattern]: https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern
+[kubebuilder-declarative-pattern-example]: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3/controllers/firstmate_controller.go
+[default-scaffold]: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3/controllers/admiral_controller.go
+[kb-issue]: https://github.com/kubernetes-sigs/kubebuilder/issues/1378
+[create-your-own-plugins]: https://deploy-preview-2115--kubebuilder.netlify.app/plugins/creating-plugins.html
+[scorecard]: ../../docs/advanced-topics/scorecard
+[kubebuilder]: https://github.com/kubernetes-sigs/kubebuilder 
```

Comment:
The `https://deploy-preview-2115--kubebuilder` requires to be replaced when the PR https://github.com/kubernetes-sigs/kubebuilder/pull/2115 gt merged.